### PR TITLE
feat: get multiple contracts method in contract cache

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Ability to fetch multiple contract types in parallel

### How I did it

ThreadPoolExecutor and adding a method to the `chain.contracts`

### How to verify it

tests

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/managers/chain.py [hunk 1]
```diff
@@ -504,6 +505,42 @@ def __getitem__(self, address: AddressType) -> ContractType:
 
         return contract_type
 
+    def get_all(
```
Comment (unparalleled-js) [inline src/ape/managers/chain.py:508]: I like `get_all` as a name, it seems to make sense considering we have `get()`.
But am open to alternatives! Just let me know or let me know what issues you have with this one.

Comment (banteg) [reply src/ape/managers/chain.py:508]: `get_multiple` or `get_many` as an alternative

**Comment by unparalleled-js** [comment]
@fubuloubu Would like to know what you are thinking on this, as it will help me make other decisions!

File: src/ape/managers/chain.py [hunk 2]
```diff
@@ -504,6 +505,44 @@ def __getitem__(self, address: AddressType) -> ContractType:
 
         return contract_type
 
+    def get_multiple(
```
Comment (unparalleled-js) [inline src/ape/managers/chain.py:508]: Could combine this with `get()` but that makes the `default=` argument kind of strange so a separate method made more sense.

Comment (fubuloubu) [reply src/ape/managers/chain.py:508]: Let's just go with it for now.

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**